gqlalchemy/query_builders/memgraph_query_builder.py

```diff
@@ -79,8 +80,112 @@ def load_csv(self, path: str, header: bool, row: str) -> "DeclarativeBase":
+
+        Args:
+            node_labels: If string, used for both sides of the relationship, if list of two, used in order.
+            relationship_types: A string or iterable of types of relationships used in the subgraph.
+            relationship_direction: Enum representing direction.
+
+        Returns:
+            a string representing a MATCH query for a path with given node labels and relationship types.
+        """
+        if not node_labels or isinstance(node_labels, str):
+            first_node = second_node = CypherNode(node_labels)
+        else:
+            if len(node_labels) > 2:
+                raise ValueError("only two node labels are permitted, one for outgoing node and one for incoming node.")
+            first_node = CypherNode(node_labels[0])
+            second_node = CypherNode(node_labels[1])
+
+        query = f" MATCH p={first_node}{CypherRelationship(relationship_types, relationship_direction)}{second_node}"
+
+        return query
+
+    def call(
+        self,
+        procedure: str,
+        arguments: Optional[Union[str, Tuple[Union[str, int, float]]]] = None,
+        node_labels: Optional[Union[str, Iterable[str]]] = None,
+        relationship_types: Optional[Union[str, Iterable[str]]] = None,
+        relationship_direction: Optional[RelationshipDirection] = RelationshipDirection.RIGHT,
+        subgraph_query: str = None,
+    ) -> "DeclarativeBase":
+        """Override of base class method to support Memgraph's subgraph functionality.
+
+        Method can be called with node labels and relationship types, both being optional, which are used to construct
+        a subgraph, or if neither is provided, a subgraph query is used, which can be passed as a string representing a
+        Cypher query defining the MATCH clause which selects the nodes and relationships to use.
+
+        Args:
+            procedure: A string representing the name of the procedure in the
+              format `query_module.procedure`.
+            arguments: A string representing the arguments of the procedure in
+              text format.
+            node_labels: Labels that define the subgraph.
+            relationship_types: Types of relationships to be used in the subgraph.
+            relationship_direction: Direction of the relationship.
+            subgraph_query: Optional way to define the subgraph via a Cypher MATCH clause.
+
+        Returns:
+            A `DeclarativeBase` instance for constructing queries.
+
+        Examples:
+            Python: `call('export_util.json', '/home/user', "LABEL", ["TYPE1", "TYPE2"]).execute()
+            Cypher: `MATCH p=(:LABEL)-[:TYPE1 | :TYPE2]->(:LABEL) WITH project(p) AS graph
+                    CALL export_util.json(graph, '/home/user')`
+
+            or
+
+            Python: `call('export_util.json', '/home/user', subgraph_query="MATCH (:LABEL)-[:TYPE]->(:LABEL)").execute()
```

Comment:
is it easier to specify `(:LABEL)-[:TYPE]->(:LABEL)` instead of `MATCH (:LABEL)-[:TYPE]->(:LABEL)`
I guess the intention was not to use cypher so I would reason to ommit this and generate by itself

Reply:
Yup, both seem a bit bad because users should use Cypher construct `(:LABEL)-[:TYPE]->(:LABEL)` and omit the `MATCH p=` or `p=`, but this seems better. The `subgraph_query` parameter is for the long tail of uses when someone wants, for instance, a path consisting of three nodes to create a subgraph